sdk/core/Azure.Core/src/RequestConditions.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace Azure
 {
-    public class ModifiedSinceConditions : MatchConditions
+    public class RequestConditions : MatchConditions
```

Comment:
I might have missed the name change discussion, but why are we changing the name?

Reply:
Storage will be exposing this type several places in their public API.  Calling it ModifiedSinceConditions is misleading because that makes it sound like it only contains If-Modified-Since and If-Unmodified-Since precondition headers, when it actually contains all the request conditions that we support in the SDK (all four of them).

Reply:
I see. 
But, I think RequestConditions is not ideal either. It sounds like a base type of all conditions, i.e. MatchConditions is also "request conditions". Also, I was hoping for these two types to end up close to each other in reference documentation (as their names both started with "M").

It's kind of late in the product cycle, so I am ok with merging this.